database/repository/exchange/exchange.go

```diff
@@ -0,0 +1,229 @@
+package exchange
+
+import (
+	"context"
+	"database/sql"
+	"encoding/csv"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/gofrs/uuid"
+	"github.com/thrasher-corp/gocryptotrader/common/cache"
+	"github.com/thrasher-corp/gocryptotrader/database"
+	modelPSQL "github.com/thrasher-corp/gocryptotrader/database/models/postgres"
+	modelSQLite "github.com/thrasher-corp/gocryptotrader/database/models/sqlite3"
+	"github.com/thrasher-corp/gocryptotrader/database/repository"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/sqlboiler/boil"
+	"github.com/thrasher-corp/sqlboiler/queries/qm"
+)
+
+// One returns one exchange by Name
+func One(in string) (Details, error) {
+	return one(in, "name")
+}
+
+// OneByUUID returns one exchange by UUID
+func OneByUUID(in uuid.UUID) (Details, error) {
+	return one(in.String(), "id")
+}
+
+// one returns one exchange by clause
+func one(in, clause string) (out Details, err error) {
+	if database.DB.SQL == nil {
+		return out, database.ErrDatabaseSupportDisabled
+	}
+
+	whereQM := qm.Where(clause+"= ?", in)
+	if repository.GetSQLDialect() == database.DBSQLite3 {
+		ret, errS := modelSQLite.Exchanges(whereQM).One(context.Background(), database.DB.SQL)
+		if errS != nil {
+			return out, errS
+		}
+		out.Name = ret.Name
+		out.UUID, errS = uuid.FromString(ret.ID)
+		if errS != nil {
+			return out, errS
+		}
+	} else {
+		ret, errS := modelPSQL.Exchanges(whereQM).One(context.Background(), database.DB.SQL)
+		if errS != nil {
+			return out, errS
+		}
+		out.Name = ret.Name
+		out.UUID, errS = uuid.FromString(ret.ID)
+		if errS != nil {
+			return out, errS
+		}
+	}
+
+	return out, err
+}
+
+// Insert writes a single entry into database
+func Insert(in Details) error {
+	if database.DB.SQL == nil {
+		return database.ErrDatabaseSupportDisabled
+	}
+
+	ctx := context.Background()
+	tx, err := database.DB.SQL.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	if repository.GetSQLDialect() == database.DBSQLite3 {
+		err = insertSQLite(ctx, tx, []Details{in})
+	} else {
+		err = insertPostgresql(ctx, tx, []Details{in})
+	}
+
+	if err != nil {
+		errRB := tx.Rollback()
+		if errRB != nil {
+			return errRB
+		}
+		return err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		errRB := tx.Rollback()
+		if errRB != nil {
+			log.Errorln(log.DatabaseMgr, errRB)
```

Comment:
good point, Will fix all the outputs to display the original query error as well if a roll back does fail

Reply:
query errors are now passed back instead of the commit/rollback if they fail to make it easier to see errors

Example from the same seed command:

```shell
❯ go run ./cmd/dbseed -verbose -config "~/.gocryptotrader/sqlite.json" exchange default
zsh: correct 'exchange' to 'exchanges' [nyae]? n
GoCryptoTrader database seeding tool
Copyright (c) 2014-2020 The GoCryptoTrader Developers.

exchange ITBIT last manual update of available currency pairs has exceeded 30 days. Manual update required!
Database file: gct_dev
INSERT INTO "exchange" ("id","name") VALUES (?,?)
[85c560f4-d090-45dc-acfa-86c2dbd07f5e Binance]
sql: transaction has already been committed or rolled back
2020/07/24 10:39:07 sqlite3: unable to insert into exchange: no such table: exchange
exit status 1
```